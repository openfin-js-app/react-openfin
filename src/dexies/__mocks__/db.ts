import Dexie from 'dexie';
import indexedDB from 'fake-indexeddb';

Dexie.dependencies.indexedDB = indexedDB;
Dexie.dependencies.IDBKeyRange = require('fake-indexeddb/lib/FDBKeyRange');

import { IConfigDexie } from '../../reduxs';

class OpenfinReactDatabase extends Dexie{
    configs:Dexie.Table<IConfigDexie,number>;

    constructor(databaseName){
        super(databaseName);
        this.version(1).stores({
            configs:'++id,tabName,fieldName,version'
        })
    }
}

const db = new OpenfinReactDatabase('openfinReactDb');

db.open().catch((err)=>{
    console.error('Default openfinReactDb db failed',err);
});

export default db;