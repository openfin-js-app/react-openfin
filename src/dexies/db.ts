import Dexie from 'dexie';

import { IConfigDexie } from '../reduxs';

class OpenfinReactDatabase extends Dexie{
    configs:Dexie.Table<IConfigDexie,number>;

    constructor(databaseName){
        super(databaseName);
        this.version(1).stores({
            configs:'++id,tabName,fieldName'
        })
        this.version(2).stores({
            configs:'++id,tabName,fieldName,version'
        })
    }
}

const db = new OpenfinReactDatabase('openfinReactDb');

db.open().catch((err)=>{
    console.error('Default openfinReactDb db failed',err);
});

export default db;