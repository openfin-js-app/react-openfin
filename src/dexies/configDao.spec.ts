import {
    CONFIG_VERSION,
    findAll, findAllOfCurrentVersion,
    findOneField, findOneFieldVal,
    saveOrUpdateOneByTabNameFieldName
} from './configDao';
import db from './db';

jest.mock('./db');

describe('ConfigDao',()=>{

    it('findAll async',async ()=>{
        const founds = await findAll();
        expect(founds).toMatchSnapshot();
    })

    it('findAllOfCurrentVersion async',async ()=>{
        const founds = await findAllOfCurrentVersion();
        expect(founds).toMatchSnapshot();
    })

    it('findOneField async',async ()=>{
        const founds = await findOneField('tabName','fieldName');
        expect(founds).toMatchSnapshot();
    })

    it('findOneFieldVal with null async',async ()=>{
        const founds = await findOneFieldVal('tabName','fieldName');
        expect(founds).toMatchSnapshot();
    })

    it('saveOrUpdateOneByTabNameFieldName async',async ()=>{
        const saved = await saveOrUpdateOneByTabNameFieldName(
            'tabName', 'fieldName', 'value'
        );
        expect(saved).toMatchSnapshot();
    })

    describe('with preset',()=>{
        beforeAll(()=>{
            const configs = db.table('configs');
            configs.put({
                tabName:'tabName', fieldName:'fieldName', value:'value', version:CONFIG_VERSION-1,
            });
            configs.put({
                tabName:'tabName', fieldName:'fieldName', value:'value', version:CONFIG_VERSION,
            });
            configs.put({
                tabName:'tabName', fieldName:'fieldName', value:'value', version:CONFIG_VERSION,
            });
        })

        it('findAll async with preset',async ()=>{
            const founds = await findAll();
            expect(founds).toMatchSnapshot();
        })

        it('findAllOfCurrentVersion async with preset',async ()=>{
            const founds = await findAllOfCurrentVersion();
            expect(founds).toMatchSnapshot();
        })

        it('saveOrUpdateOneByTabNameFieldName async with preset',async ()=>{
            const saved = await saveOrUpdateOneByTabNameFieldName(
                'tabName', 'fieldName', 'value'
            );
            expect(saved).toMatchSnapshot();
        })


        it('saveOrUpdateOneByTabNameFieldName async update',async ()=>{
            const saved = await saveOrUpdateOneByTabNameFieldName(
                'tabName', 'fieldName', 'value2'
            );
            expect(saved).toMatchSnapshot();
        })

        it('findOneFieldVal null async',async ()=>{
            const founds = await findOneFieldVal('tabName','fieldName');
            expect(founds).toMatchSnapshot();
        })

    })

});