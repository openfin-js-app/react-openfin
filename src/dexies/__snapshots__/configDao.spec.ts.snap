// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigDao findAll async 1`] = `Array []`;

exports[`ConfigDao findAllOfCurrentVersion async 1`] = `Array []`;

exports[`ConfigDao findOneField async 1`] = `undefined`;

exports[`ConfigDao findOneFieldVal with null async 1`] = `null`;

exports[`ConfigDao saveOrUpdateOneByTabNameFieldName async 1`] = `
Object {
  "fieldName": "fieldName",
  "id": 1,
  "tabName": "tabName",
  "value": "value",
  "version": 2,
}
`;

exports[`ConfigDao with preset findAll async with preset 1`] = `
Array [
  Object {
    "fieldName": "fieldName",
    "id": 1,
    "tabName": "tabName",
    "value": "value",
    "version": 2,
  },
  Object {
    "fieldName": "fieldName",
    "id": 2,
    "tabName": "tabName",
    "value": "value",
    "version": 1,
  },
  Object {
    "fieldName": "fieldName",
    "id": 3,
    "tabName": "tabName",
    "value": "value",
    "version": 2,
  },
  Object {
    "fieldName": "fieldName",
    "id": 4,
    "tabName": "tabName",
    "value": "value",
    "version": 2,
  },
]
`;

exports[`ConfigDao with preset findAllOfCurrentVersion async with preset 1`] = `
Array [
  Object {
    "fieldName": "fieldName",
    "id": 1,
    "tabName": "tabName",
    "value": "value",
    "version": 2,
  },
  Object {
    "fieldName": "fieldName",
    "id": 3,
    "tabName": "tabName",
    "value": "value",
    "version": 2,
  },
  Object {
    "fieldName": "fieldName",
    "id": 4,
    "tabName": "tabName",
    "value": "value",
    "version": 2,
  },
]
`;

exports[`ConfigDao with preset findOneFieldVal null async 1`] = `"value2"`;

exports[`ConfigDao with preset saveOrUpdateOneByTabNameFieldName async update 1`] = `
Object {
  "fieldName": "fieldName",
  "id": 5,
  "tabName": "tabName",
  "value": "value2",
  "version": 2,
}
`;

exports[`ConfigDao with preset saveOrUpdateOneByTabNameFieldName async with preset 1`] = `
Object {
  "fieldName": "fieldName",
  "id": 5,
  "tabName": "tabName",
  "value": "value",
  "version": 2,
}
`;
